Issue #29: Get CA COVID-19 Hospital Data

benghancock opened the pull request:
This is a PR to add a script to fetch hospitalization data from the data.ca.gov API, raised in issue # 29. I've tried to keep the output model as consistent as possible with the existing model, although there are necessarily some differences. I've also added some documentation of the output model in the README. I included logging for easier debugging/transparency, but that can be removed for consistency. Let me know if you spot bugs or other issues.

benghancock commented:
Looks like I have some issues picked up by the linter to address -- I'll work on fixing those

elaguerta commented on data_models/README.md:
````diff
@@ -136,8 +136,48 @@ The fields will be used for normalizing the county case and death tabulations, a
+        }
+```
+
+The `name` value starts with "Hospitalization - " and then the full name of the county (e.g. "Hospitalization - Alameda County"). If the argument `"all"` is passed to `get_county()` (the default in the `get_timeseries()` function), then data for all counties will be fetched, and the name value will be "Hospitalization - All CA Counties"
````
Minor issue: `get_county` passes `name` exactly as is, so for example, if you run it on `marin` it will store `name` as "Hospitalization - marin County". Could you ensure titlecase capitalization?

benghancock replied:
Yep - will handle that!

elaguerta commented on data_models/README.md:
```diff
@@ -136,8 +136,48 @@ The fields will be used for normalizing the county case and death tabulations, a
+            "hospitalized_covid_confirmed_patients": 1.0,
+            "_id": 1,
+            "all_hospital_beds": null,
+            "todays_date": "2020-03-29T00:00:00"
```
Over in [`data_models/data_model.json`](https://github.com/sfbrigade/data-covid19-sfbayarea/blob/master/data_models/data_model.json) for each timeseries, we are just storing the date part of the timestamp. I think it makes sense to do that here since it appears the data resolution is only to do the day(`T00:00:00`). Please truncate the timestamp to the date. 
 
(Are these are the correct report dates PDT given the timezone of the database? I don't know anything about timezones except to beware, so forgive my paranoia!)

Finally minor issue: I'd like to change the `"todays_date"` field name to `"report_date"`. It's clear from the metadata, but I still felt some cognitive effort scanning through the data and seeing all these dates that are not today's date! Feel free to ignore if you think it's clear enough.

benghancock replied:
> Over in [`data_models/data_model.json`](https://github.com/sfbrigade/data-covid19-sfbayarea/blob/master/data_models/data_model.json) for each timeseries, we are just storing the date part of the timestamp. I think it makes sense to do that here since it appears the data resolution is only to do the day(`T00:00:00`). Please truncate the timestamp to the date.

Got it - will do.

> (Are these are the correct report dates PDT given the timezone of the database? I don't know anything about timezones except to beware, so forgive my paranoia!)

I can only assume that the dates in the series reflect the date on Pacific time, given that this data is for California. On the landing page for this data, the "Last Updated" time is given in UTC-07:00, so that seems like a safe assumption.

> Finally minor issue: I'd like to change the `"todays_date"` field name to `"report_date"`. It's clear from the metadata, but I still felt some cognitive effort scanning through the data and seeing all these dates that are not today's date! Feel free to ignore if you think it's clear enough.

That makes sense. I'll rename the key.

elaguerta commented on data_models/README.md:
```diff
@@ -136,8 +136,48 @@ The fields will be used for normalizing the county case and death tabulations, a
+        {
+            "icu_covid_confirmed_patients": 0.0,
+            "icu_suspected_covid_patients": 0.0,
+            "hospitalized_covid_patients": null,
```
Please replace `null` values with `-1`, per [the data README](https://github.com/sfbrigade/data-covid19-sfbayarea/blob/master/data_models/README.md#non-number-values)
(By the way! We have _another_ README in `./data_models` just for data sleuths like yourself!)

I am also tempted to ask that you cast all integer values to `int`s so they appear as whole numbers and throw errors otherwise, since that is what we are doing with other integer data. At the very least, it will help the front end be consistent if they use tooltips to read off values. Let me know if you see a reason to keep the precision to a 1/10th.

benghancock replied:
Got it. All of the data in the fields I've seen so far (except for `rank`) are either whole numbers or null, so casting as `int` makes sense. I'll follow up if I find it's throwing lots of errors.

elaguerta replied:
What is `rank`? I just noticed that. You don't mention it in the README additions nor in `hospitals_data_model.json`. Based on a quick visual inspection of Marin's timeseries, it looks like it's the same value for every day.

Do you know what the `rank` column is - if so, can you add descriptions to the README and hospitals data model? If not, I'm inclined to skip the column for now. You can flag it in the README for the hospitalization section, and also link to a new issue to resolve it when we know more.

benghancock replied:
That field appears to be undocumented, and so I can only guess as to what it is. I'll remove it from the data, and as you suggest, open a new issue and make a note in the README.

benghancock commented:
Thanks @elaguerta! See my inline responses above. As far as updating `scraper_data.py`, I'd be happy to do that. I do think it makes sense to put that in as a separate PR to chunk up the work. I'll work on these issues and should be able to wrap up in the next couple days.

benghancock commented:
I've made the changes requested above, though still need to reflect those in the data README. I'll work on that next.

benghancock commented:
Hi @elaguerta I think I've addressed all the requested changes, and added documentation about the data model. Let me know if you spot anything else. Otherwise, I'll start working on integrating this into `scraper_data.py` as discussed above. Thanks!

elaguerta commented:
Hi @benghancock, thanks for making all requested changes. I left one more request inline regarding`rank`. After that, this looks good to go!

benghancock commented:
Thanks @elaguerta! That's done -- see my response to your inline comment above. I'll turn to integrating this into the scraper.
